README.md

```diff
@@ -1,100 +1,143 @@
 ```
 
-_Note: You should now see the name of the virtual environment to the left of the prompt._
+### Virtual Environment Setup
 
-### Install App/Package
+You can explicitly create a virtual environment by running the following command:
 
-To install the package/bundle/application:
+```python
+$ python -m venv .venv
+```
+
+To activate the `venv` virtual environment and also retreive the path to your virtual environment, run:
 
 ```bash
-pip[3] install -e .
+$ source $(poetry env info --path)/bin/activate
 ```
 
-To verify that the package has been installed (as a `pip` package), check here:
+Note: If you don't want to create a `venv` virtual environment, `poetry` will create a virtual environment for you. To use `poetry`'s virtual environment, you can skip this step.
+
+### Install Dependencies
+
+Running the following command reads the `pyproject.toml` file from the current project, resolves the dependencies and install them:
 
 ```bash
-pip[3] list
+$ poetry install
 ```
 
-Now, your environment is ready to test the modules within the application.
+### Obtain Credentials File(s)
 
-Once you have finished testing the application within the virtual environment and want to deactivate it, simply run:
+Download a copy of the `credentials.json` file stored on Synapse by running the following command:
 
 ```bash
-deactivate
+$ synapse get syn21088684
 ```
 
-To run any of the example file(s), go to your root directory and execute/run python script in the following way:
-
-Let's say you want to run the `metadata_usage` example - then do this:
+Alternatively, you can also download the `schematic_service_account_creds.json` file, which uses the service account mode of authentication (_for Google services_). To download this file, run the following:
 
 ```bash
-python[3] examples/metadata_usage.py
+$ synapse get syn24214983
 ```
 
-### Configure Synapse Credentials
+Note: The service account mode of authentication is still being tested and certain functionalities (for our app) which leverage Google Apps Script API are [not fully compatible with this mode](https://developers.google.com/apps-script/api/concepts). So we recommend you follow the first command in this section.
 
-Download a copy of the `credentials.json` file (or the file needed for authentication using service account, called `quickstart-1560359685924-198a7114b6b5.json`) stored on Synapse, using the synapse client command line utility. The credentials file is necessary for authentication to use Google services/APIs. To do so:
+When you are using the Google credentials in the `credentials.json` file for authentication, you will be prompted to download a `token.pickle` file when you access `schematic` services (via the command line) for the first time. It will download the `token.pickle` file to the project root by default. There are configurable keys/attributes for both of these files in the `config.yml` file.
```

Comment:
Apologies I missed responding to this @xdoan. By "using the credentials for authentication" I/we simply mean using the `credentials.json` file to create a `token.pickle` file if one doesn't exist already. The `token.pickle` will then be used when we make repeated calls to Google APIs/services (via the Googlesheets/Google Drive service objects). You can see the above behaviour in the [`schematic.utils.google_api_utils.build_credentials`](https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/utils/google_api_utils.py#L18) method.

Reply:
I think you should include the information you explained to me in the comment and be more explicit so people understand when and how this would be used. I find myself needing to explain this to people who are using schematic and its confusing for them. Rong tried this a few weeks ago (maybe it wasn't this version yet) but schematic didn't seem to prompt him to authenticate his gsheets API and he had to do it manually by calling the quickstart.  

Points I would like to know as a naive user to clarify: 

- when this should happen

- if it doesn't happen what to do

- how it happens; what are the exact steps involved so I know I did it correctly(e.g. you will authenticate through google APIs with a google account (which you may need login info for) and then a token.pickle is automatically downloaded to your root is more explicit than, `you will be prompted to download a token.pickle` which is perhaps an oversimplification).
Rong also had some confusion because the browser tried to block the API and he had to click "advanced" to continue the auth process, which is why I think being more explicit would be more helpful (we don't need to walk them through the google logging in part, but I think its useful to share the steps involved)

Reply:
Thanks @xdoan. That makes sense yes, I can understand how the `README` wasn't sufficiently clear in detailing the purpose(s) of `credentials.json` and `token.pickle` files. I've added a [new commit](https://github.com/Sage-Bionetworks/schematic/pull/392/commits/8f155f4dde28d1a818fc8a52c37a9d22cc8c0c16) which addresses this. Could you go over it and tell me if I'm still missing something?

Reply:
looks good, thanks for adding more detail!